File: Bridge/System/Threading/Semaphore.cs
using Bridge;

namespace System.Threading
{
    public sealed class Semaphore : WaitHandle
    {
        private Semaphore(IntPtr handle)
        {
        }

        public Semaphore(int initialCount, int maximumCount)
            : this(initialCount, maximumCount, null)
        {
        }

        public Semaphore(int initialCount, int maximumCount, string name)
        {
        }
        
        public int Release()
        {
            throw new NotImplementedException();
        }

        public int Release(int releaseCount)
        {
            throw new NotImplementedException();
        }
        
        public static Semaphore OpenExisting(string name)
        {
            throw new NotImplementedException();
        }
    }
}

File: Bridge/Resources/.generated/system/threading/semaphore.js
    Bridge.define("System.Threading.Semaphore", {
        inherits: [System.Threading.WaitHandle],
        statics: {
            methods: {
                openExisting: function (name) {
                    throw new System.NotImplementedException();
                }
            }
        },
        ctors: {
            $ctor2: function (handle) {
                this.$initialize();
                System.Threading.WaitHandle.ctor.call(this);
            },
            ctor: function (initialCount, maximumCount) {
                System.Threading.Semaphore.$ctor1.call(this, initialCount, maximumCount, null);
            },
            $ctor1: function (initialCount, maximumCount, name) {
                this.$initialize();
                System.Threading.WaitHandle.ctor.call(this);
            }
        },
        methods: {
            release: function () {
                throw new System.NotImplementedException();
            },
            release$1: function (releaseCount) {
                throw new System.NotImplementedException();
            }
        }
    });
